src/stackFrame_loongarch64.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 Andrei Pangin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifdef __loongarch_lp64
+
+#include <errno.h>
+#include <string.h>
+#include <sys/syscall.h>
+#include "stackFrame.h"
+
+#define REG(l)  _ucontext->uc_mcontext.__gregs[l]
+
+uintptr_t& StackFrame::pc() {
+    return (uintptr_t&)_ucontext->uc_mcontext.__pc;
+}
+
+uintptr_t& StackFrame::sp() {
+    return (uintptr_t&)REG(LARCH_REG_SP);
+}
+
+uintptr_t& StackFrame::fp() {
+    return (uintptr_t&)REG(22);
+}
+
+uintptr_t& StackFrame::retval() {
+    return (uintptr_t&)REG(LARCH_REG_A0);
+}
+
+uintptr_t StackFrame::link() {
+    return (uintptr_t)REG(LARCH_REG_RA);
+}
+
+uintptr_t StackFrame::arg0() {
+    return (uintptr_t)REG(LARCH_REG_A0);
+}
+
+uintptr_t StackFrame::arg1() {
+    return (uintptr_t)REG(LARCH_REG_A0 + 1);
+}
+
+uintptr_t StackFrame::arg2() {
+    return (uintptr_t)REG(LARCH_REG_A0 + 2);
+}
+
+uintptr_t StackFrame::arg3() {
+    return (uintptr_t)REG(LARCH_REG_A0 + 3);
+}
+
+uintptr_t StackFrame::jarg0() {
+    return arg1();
+}
+
+uintptr_t StackFrame::method() {
+    return (uintptr_t)REG(20);
+}
+
+uintptr_t StackFrame::senderSP() {
+    return (uintptr_t)REG(26);
```

Comment:
Where do these numbers come from? Can you provide a pointer to the sources?

Reply:
It comes from [LoongArch ELF ABI specification](https://loongson.github.io/LoongArch-Documentation/LoongArch-ELF-ABI-EN.html) and I use the third Static registers `$s3` AKA `$r26` as sender SP.

Please point out my fault.

Thanks,
Leslie Zhai

Reply:
Thanks. I mean, where in JDK code r20 and r26 are mapped to method and sender_sp respectively.

Reply:
Thanks for pointing out my fault!

[Rmethod](https://github.com/loongson/jdk11u/blob/master-ls/src/hotspot/cpu/loongarch/register_loongarch.hpp#L172) should be `$r26` and [Rsender](https://github.com/loongson/jdk11u/blob/master-ls/src/hotspot/cpu/loongarch/register_loongarch.hpp#L173) shoule be `$r27`.

Thanks,
Leslie Zhai